src/adagrams.js

```diff
@@ -1,15 +1,119 @@
+// import { toPrimitive } from "core-js/fn/symbol";
+
+const LETTERPOOL = [
+  'A','A','A','A','A','A','A','A','A',
+  'B','B',
+  'C','C',
+  'D','D','D','D',
+  'E','E','E','E','E','E','E','E','E','E','E','E',
+  'F','F',
+  'G','G','G',
+  'H','H',
+  'I','I','I','I','I','I','I','I','I',
+  'J',
+  'K',
+  'L','L','L','L',
+  'M','M',
+  'N','N','N','N','N','N',
+  'O','O','O','O','O','O','O','O',
+  'P','P',
+  'Q',
+  'R','R','R','R','R','R',
+  'S','S','S','S',
+  'T','T','T','T','T','T',
+  'U','U','U','U',
+  'V','V',
+  'W','W',
+  'X',
+  'Y','Y',
+  'Z'
+]
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let letters = [];
+  let letterPoolCopy = [...LETTERPOOL];
+  while (letters.length < 10) {
+    const random = Math.floor(Math.random() * letterPoolCopy.length);
+    letters.push(letterPoolCopy[random]);
+    letterPoolCopy.splice(random, 1);
+  }
+  return letters;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let inputLetters = input.toUpperCase().split('');
+  let returnLetters = [];
+  for (const letter of inputLetters) {
+    if (lettersInHand.includes(letter)) {
+      returnLetters.push(letter);
+      lettersInHand.splice(lettersInHand.indexOf(letter), 1);
+    }
+  }
+  return inputLetters.join('') === returnLetters.join('');
```

Comment:
This works and looks readable, Well done. However You can do this in a more efficient manner by using one if statement that returns false if the user's letter is not in the letters in hand. You can also simply return true in the function and this would also work because the condion would only check for invalid words. Essentially you always return true if the user's input is valid, if not; we move into the conditional:

...
  // Loop through the letters in the user's input
  for (let i = 0; i < input.length; i++) {
    // If the user's letter is not in the letters in hand
    if (lettersInHandCopy.includes(input_array[i])) {
      // Remove the letter from the letters in the hand
      // The shift() method in JavaScript is used to remove the first element from an array and return that element.
      lettersInHandCopy.shift(input_array[i]);
    // If the user's letter is not in the letters in hand
    } else {
      return false;
    }
  }
  // If the user's input is valid
  return true;
};

Remember to keep in mind where certain built-in function calls might be hiding a loop, and try to avoid using those within an outer loop (things like `splice` or `includes`). Most of these built in functions have a 0(N) time complexity. The time complexity of JavaScript's `splice()` method is O(N), where n is the number of elements in the array. This is because it can potentially shift each element in the array per call. Therefore, it is a linear time complexity. The algorithm in this function uses a nested loop because of the functions nested in the `for/of`. So the inner loop (`indexOf()`) looks at each element for every element in the outer loop (splice). Therefore the algorithm has quadratic time complexity which makes it quite slow.